modin/core/dataframe/pandas/partitioning/partition_manager.py

```diff
@@ -67,12 +72,11 @@ def wait(*args, **kwargs):
             # serially kick off all the deferred computations so that they can
             # all run asynchronously, then wait on all the results.
             [part.drain_call_queue() for part in partitions.flatten()]
-            # need to go through all the values of the map iterator
-            # since `wait` does not return anything, we need to explicitly add
-            # the return `True` value from the lambda
-            # TODO(https://github.com/modin-project/modin/issues/4491): Wait
-            # for all the partitions in parallel.
-            all(map(lambda partition: partition.wait() or True, partitions.flatten()))
+            # The PandasOnRay partition manager invokes ray.get under the hood, which waits
+            # in parallel for all computations to finish
+            PandasOnRayDataframePartitionManager.get_objects_from_partitions(
```

Comment:
Note that we also need an implementation for the dask partition manager.

Reply:
Is it just Ray and Dask, or does this code need to be more general?

Reply:
Every class that subclasses `PandasDataframePartitionManager` needs an implementation. As far as I can tell, there's `PandasOnDaskDataframePartitionManager`,`PandasOnPythonDataframePartitionManager`, `GenericRayDataframePartitionManager` and 3 of its subclasses (but maybe you can implement this in `GenericRayDataframePartitionManager`) and `OmnisciOnNativeDataframePartitionManager`. For Python and Omnisci and Dask you can get an idea of the implementation by looking at how the `_partition_class` implements `wait`.